chore!: acir serialization changes

guipublic opened the pull request:
This PR update ACIR deserialisation to matche the ACIR changes in noir PR https://github.com/noir-lang/noir/pull/8134

TomAFrench commented:
I'm going to edit the base of this branch to `next` as we'll need to merge it into there.

dbanks12 reviewed (commented):
Change to AVM transpiler looks good to me!

TomAFrench commented on yarn-project/constants/src/constants.gen.ts:
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable */
+ 
```
Not sure why this was removed?

TomAFrench commented on barretenberg/cpp/src/barretenberg/dsl/acir_format/recursion_constraint.hpp:
Looks like this isn't hooked up to use dummy values if the predicate is false.

guipublic committed:
chore!: acir serialization changes

This PR update ACIR deserialisation to matche the ACIR changes in noir PR https://github.com/noir-lang/noir/pull/8134

Co-authored-by: Tom French <15848336+TomAFrench@users.noreply.github.com>
